src/Npgsql/TypeHandlers/TextHandler.cs

```diff
@@ -114,7 +115,9 @@ static async ValueTask<string> ReadLong(NpgsqlReadBuffer buf, int byteLen, bool
                     }
                     break;
                 }
-                return buf.TextEncoding.GetString(tempBuf);
+                return buf.UseStringCaching ?
```

Comment:
Note that this is the "bad" code path, where the string is big and doesn't fit into our buffer. The expectation is for the vast majority of cases to exit [above](https://github.com/dotnet/efcore/pull/23905/files) - you probably want to do your caching there.

Reply:
That should go via `NpgsqlReadBuffer.ReadString` which I've already added to here?

Reply:
Right, missed that you did that. Not sure it's a good idea to do caching for the big strings BTW, but this is experiment mode regardless...

Reply:
If they are beyond a certain size it just encodes it rather than trying to cache it; also single use (unique) strings as discarded from the cache first; so there shouldn't be much downside (I think...)